augur/io/metadata.py

```diff
@@ -88,3 +97,228 @@ def read_metadata(metadata_file, id_columns=("strain", "name"), chunk_size=None)
         metadata_file,
         **kwargs
     )
+
+
+def read_metadata_to_dict(metadata, duplicate_reporting=DataErrorMethod.ERROR, id_column=None):
+    """
+    Read rows from *metadata* file and yield each row as a single dict.
+
+    Will report duplicate records based on the *id_column* if requested via
+    *duplicate_reporting* after the generator has been exhausted.
+
+    Parameters
+    ----------
+    metadata: str
+        Path to a CSV or TSV metadata file or IO buffer
+
+    duplicate_reporting: DataErrorMethod, optional
+        How should duplicate records be reported
+
+    id_column: str, optional
+        Name of the column that contains the record identifier used for reporting duplicates.
+        Uses the first column of the metadata if not provided.
+
+    Yields
+    ------
+    dict:
+        The parsed row as a single record
+
+    Raises
+    ------
+    AugurError:
+        When there are parsing errors from the csv standard library or when
+        requested to error on duplicate records
```

Comment:
If you want to be explicit:

```suggestion
        Raised for any of the following:
        1. There are parsing errors from the csv standard library
        2. The duplicate reporting method is set to error and duplicate(s) are found
        3. An unhandled duplicate reporting method is specified
```

Reply:
Fixed up in d56388295fcc804fbe42d474b2e3636f92b251e5